assets/hip-1037/Specification-Format-Style-Guidelines.md

```diff
@@ -0,0 +1,408 @@
+   - In field comments a `<p>` tag is _required_ for paragraph breaks.
+     Blank lines break the automated documentation generator in **fields**
+     because fields are output as markdown tables, and markdown tables require
+     each row to be a single long line.
+- Be _prescriptive_ in requirements.  Do not _describe_ the behavior.
+   - Declare what MUST be done for valid inputs, and what SHALL be present in
+     valid outputs (or in state messages).
+   - Declare what SHALL happen as a result of a successful, or failed,
+     transaction.
+   - The word `will` usually means `SHALL` or `MUST`.
+   - The word `is` often means `SHALL be`.
+   - Document both positive (`SHALL` or `MUST`) and negative (`SHALL NOT` or
+     `MUST NOT`) requirements.
+   - Avoid "perfect" or "imperfect" verb aspects and "wiggle" words
+     (might, could, perhaps, etc...).
+      - The specification items MAY, SHOULD, and RECOMMENDED serve to fill
+        the _occasional_ need for less imperative requirements.
+   - Also avoid "progressive" verbs (e.g. working, writing, going, deleting).
+- Requirements should be short and focused.
+   - Better to have three separate lines for three items than one longer
+     line with three clauses.
+   - Duplicating text across lines with slightly different requirements
+     is not only acceptable, but often recommended.
+   - Well written specification tends to be much more short and clipped
+     sentences than "normal" prose. Some describe the flow as "staccato".
+- Required and Optional fields
+   - The default state for any field is "optional", and all required fields
+     should be specified as `REQUIRED`. If the presence or absence of a field
+     must be clear _separate_ from the default value of that field, that field
+     should use proto3 "explicit presence" and be marked with
+     the `optional` keyword.
+- Add another blank line after requirements for `message` documents only.
+
+### Other elements
+- Each message that represents a transaction body **must** document the
+  Block Stream effects with a heading `### Block Stream Effects`.
+- General notes go last, for `message` only, with a heading
+  `#### Additional Notes`.
+- There are cases where description text for a message or field is
+  exceptionally detailed. In these cases it _may_ be appropriate to
+  separate broad categories of information with a single horizontal line
+  (i.e. an `<hr>` tag, `---` in markdown). These cases _should_ be quite
+  rare, however; we prefer to avoid the need for such large blocks of
+  specification text.
+- Line oriented comments (i.e. lines prefixed with `//`) may be used for
+  any comments needed in a proto file that are not intended for API
+  specification.  Examples include explaining an `oneof` block, or describing
+  why a particular field number or name is reserved.
+
+## Content Guidelines
+### General `proto` File Guidelines
+- Protocol buffers are compiled for the `hedera-services` codebase using a
```

Comment:
Q: Is this true? Isn't it applicable network wide for MN, BN, SDK repos etc and any 3rd party users?

Reply:
PBJ is only used by Services at the moment; eventually Block Node will be added.  I do not know if MN uses PBJ, but I do not _think_ so.
SDKs do not use PBJ, only the Java SDK _could_ and it _should not_ for compatibility with the widest array of users.
3rd party systems written in Java _might_ choose to use PBJ, but I strongly doubt any currently are, and there are good reasons not to (not least that PBJ is not a complete protoc replacement, but is quite strongly focused on the CN use cases).